ts/src/test/Exchange/base/test.sharedMethods.ts

```diff
@@ -40,7 +40,7 @@ function assertType (exchange: Exchange, skippedProperties: object, entry: objec
     return result;
 }
 
-function assertStructure (exchange: Exchange, skippedProperties: object, method: string, entry: object, format: any[] | object, emptyAllowedFor = []) {
+function assertStructure (exchange: Exchange, skippedProperties: object, method: string, entry: object, format: any[] | object, emptyAllowedFor: any [] = undefined, deep = false) {
```

Comment:
do we need this deep flag?

Reply:
I don\t understand when we wouldn\t do a deep comparison

Reply:
let me check

Reply:
ok, so it is needed ***at this moment***, because otherwise we get 
![image](https://github.com/user-attachments/assets/914ac05c-19dc-45e6-8911-616a8f4d2772)
it's in market structure, where we have such sample format :
```
        'precision': {
            'price': exchange.parseNumber ('0.001'), // integer or fraction
            'amount': exchange.parseNumber ('0.001'), // integer or fraction
            'cost': exchange.parseNumber ('0.001'), // integer or fraction
        },
```

so, in sample we have `cost` but it expected that all exchanges markets>precision should have cost. so, to avoid the delays for this PR (at deep checking other things is not direct scope of this PR) I would leave `deep` flag for now. 
in separate PRs i'll find out a way to handle other deep-check issues